mysql-test/suite/innodb/t/percona_changed_page_bmp_requests_debug.test

```diff
@@ -25,3 +26,133 @@ SET @@GLOBAL.innodb_log_checkpoint_now=TRUE;
+--source include/restart_mysqld.inc
+
+--let $assert_text = InnoDB redo log tracking must be enabled
+--let $assert_cond = @@innodb_track_changed_pages = 1
+--source include/assert.inc
+
+--let $assert_text = InnoDB file per table must be disabled
+--let $assert_cond = @@innodb_file_per_table = 0
+--source include/assert.inc
+
+--echo
+--echo # Creating a simple table and filling it with some records.
+CREATE TABLE t1(id INT) ENGINE=InnoDB STATS_PERSISTENT=0 STATS_AUTO_RECALC=0;
+INSERT INTO t1 VALUES (1), (2), (3), (4), (5), (6), (7), (8), (9), (10);
+
+--echo
+--echo # Flushing the table and changed page bitmaps.
+FLUSH TABLES t1;
+SET @@GLOBAL.innodb_log_checkpoint_now=TRUE;
+SET @@GLOBAL.innodb_track_redo_log_now=TRUE;
+
+--echo
+--echo # Inserting and flushing one more time for compatibility with 5.5.
+INSERT INTO t1 VALUES (1), (2), (3), (4), (5), (6), (7), (8), (9), (10);
+FLUSH TABLES t1;
+SET @@GLOBAL.innodb_log_checkpoint_now=TRUE;
+SET @@GLOBAL.innodb_track_redo_log_now=TRUE;
+
+--echo
+--echo # Listing .xdb files after the flush.
+--echo # The output must show three .xdb files: ib_modified_log_1_0.xdb, ib_modified_log_2_nnn.xdb and ib_modified_log_3_nnn.xdb.
+--replace_regex /_[[:digit:]]+\.xdb$/_<nnn>.xdb/
+--list_files $CUSTOM_INNODB_DATA_HOME_DIR *.xdb
+
+--echo
+--echo # Purging changed page bitmaps (tracking enabled).
+--echo # The first two .xdb files (ib_modified_log_1_0.xdb and ib_modified_log_2_nnn.xdb) must be deleted.
+--let $previous_lsn = `$select_lsn_query`
+--replace_regex /[[:digit:]]+/PREVIOUS_LSN/
+eval PURGE CHANGED_PAGE_BITMAPS BEFORE $previous_lsn;
+
+--echo
+--echo # Listing .xdb files after the purge (tracking enabled).
+--echo # The output must show only one .xdb file - ib_modified_log_3_nnn.xdb.
+--replace_regex /_[[:digit:]]+\.xdb$/_<nnn>.xdb/
+--list_files $CUSTOM_INNODB_DATA_HOME_DIR *.xdb
+
+--echo
+--echo # Inserting a few more records into the table.
+INSERT INTO t1 VALUES (1), (2), (3), (4), (5), (6), (7), (8), (9), (10);
+
+--echo
+--echo # Flushing the table and changed page bitmaps again.
+FLUSH TABLES t1;
+SET @@GLOBAL.innodb_log_checkpoint_now=TRUE;
+SET @@GLOBAL.innodb_track_redo_log_now=TRUE;
+
+--echo
+--echo # Listing .xdb files after the second flush.
+--echo # The output must show two .xdb files: ib_modified_log_3_nnn.xdb and ib_modified_log_4_nnn.xdb.
+--replace_regex /_[[:digit:]]+\.xdb$/_<nnn>.xdb/
+--list_files $CUSTOM_INNODB_DATA_HOME_DIR *.xdb
+
+--echo
+--echo # Restarting with redo log tracking disabled.
+--let $restart_parameters = restart:--innodb-track-changed-pages=0 --innodb-max-bitmap-file-size=4096 --innodb-data-home-dir=$CUSTOM_INNODB_DATA_HOME_DIR --innodb-log-group-home-dir=$CUSTOM_INNODB_DATA_HOME_DIR --innodb-file-per-table=0
+--source include/restart_mysqld.inc
+
+--let $assert_text = InnoDB redo log tracking must be disabled
+--let $assert_cond = @@innodb_track_changed_pages = 0
+--source include/assert.inc
+
+--let $assert_text = InnoDB file per table must be disabled
```

Comment:
I added `--innodb-file-per-table=0` restart parameter to 5.6 and 5.7 (both to the first restart with tracking enabled and to the second one with tracking disabled) just to have the same settings as in 5.5 (in which it is OFF by default).

My reasoning for that is that `$select_lsn_query`, which determines the LSN of the last `.xdb` file, relies on the `space_id` being `0`. And I thought that with `--innodb-file-per-table=1` this would no longer be true.

Reply:
I understand why it's needed when tracking is enabled, but why it's needed for this one restart where tracking is disabled?

Reply:
In the previous restart we had
```
--innodb-track-changed-pages=1
--innodb-max-bitmap-file-size=4096
--innodb-data-home-dir=$CUSTOM_INNODB_DATA_HOME_DIR
--innodb-log-group-home-dir=$CUSTOM_INNODB_DATA_HOME_DIR
--innodb-file-per-table=0
```
I just wanted to keep the changes in restart parameters minimal and leave everything else as it was in the previous restart - so changed only `--innodb-track-changed-pages=1` to `--innodb-track-changed-pages=0` here.

If you think this is misleading, I can change it.

Reply:
Assert for it might be misleading, I'd just remove the option from this restart and the assert.